### PR TITLE
[SPARK-8776] Increase the default MaxPermSize

### PR DESCRIPTION
**Pull request by yhuai**
I am increasing the perm gen size to 256m.

https://issues.apache.org/jira/browse/SPARK-8776

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -206,7 +206,7 @@ object SparkBuild extends PomBuild {
     fork := true,
     outputStrategy in run := Some (StdoutOutput),
 
-    javaOptions ++= Seq("-Xmx2G", "-XX:MaxPermSize=1g"),
+    javaOptions ++= Seq("-Xmx2G", "-XX:MaxPermSize=256m"),
```
Comment (yhuai) [inline project/SparkBuild.scala]: @marmbrus This setting only affect sparkShell and sparkSql, right?

Comment (marmbrus) [inline project/SparkBuild.scala]: and probably the tests

Comment (andrewor14) [inline project/SparkBuild.scala]: wait, why do we want to lower it here?

Comment (marmbrus) [inline project/SparkBuild.scala]: We just want to align it with what happens when you run the spark-shell normally.

**Comment by yhuai** [comment]
test this please.

**Comment by andrewor14** [comment]
retest this please

File: launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java [hunk 2]
```diff
@@ -194,7 +194,7 @@ private void testCmdBuilder(boolean isDriver) throws Exception {
         if (isDriver) {
           assertEquals("-XX:MaxPermSize=256m", arg);
         } else {
-          assertEquals("-XX:MaxPermSize=128m", arg);
+          assertEquals("-XX:MaxPermSize=256m", arg);
```
Comment (yhuai) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: @andrewor14 What does `isDriver` do at here?

Comment (andrewor14) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: I think it literally just says give more perm space for the driver than executors. This change is probably OK.

**Comment by yhuai** [comment]
OK. I am merging it to master and branch 1.4.

File: launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java [hunk 3]
```diff
@@ -194,7 +194,7 @@ private void testCmdBuilder(boolean isDriver) throws Exception {
         if (isDriver) {
```
Comment (srowen) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: You can remove this condition then?

Comment (andrewor14) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: oops, yes

Comment (srowen) [inline launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java]: Probably not worth another PR since it's already merged

**Pull request closed without merging**